Tests/Services/ParkingSlotServiceTests.cs

```diff
@@ -305,5 +308,83 @@ public void GivenSlotWithReservations_WhenIsSlotFreeIsCalled_ThenExpectedResultR
             Assert.Equal(result, expectedResult);
         }
         #endregion
+
+        #region Filter
+        public void GivenFilterSlotVmWithAllCategory_WhenFilterIsCalled_ThenReturnedFilteredSlotsFilteredByIsFreeProperty()
+        {
+            //Arrange
+            var slotVM = new FilterSlotVM()
+            {
+                ZoneId = _testZoneId,
+                Category = 0,
+                OnlyFree = true
+            };
+
+            var filterSlotQuery = new FilterSlotsQuery(slotVM);
+            var slots = new List<ParkingSlot>()
+            {
+                _testSlot,
+                new()
+                {
+                    ParkingZoneId = _testZoneId,
+                    Category = SlotCategoryEnum.Business,
+                    Reservations = new Collection<Reservation>()
+                    {
+                        new()
+                        {
+                            StartTime = DateTime.Now.AddHours(-1),
+                            Duration = 2
+                        }
+                    }
+                }
+            };
+            var expectedSlot = new List<ParkingSlot>()
+            {
+                _testSlot
+            };
+
+            //Act
+            var result = service.Filter(filterSlotQuery);
+
+            //Assert
+            Assert.Equal(JsonSerializer.Serialize(expectedSlot), JsonSerializer.Serialize(result));
+        }
+
+        public void GivenFilterSlotVmWithStandardCategory_WhenFilterIsCalled_ThenReturnedFilteredSlotsFilteredByIsFreeProperty()
+        {
+            //Arrange
+            var slotVM = new FilterSlotVM()
+            {
+                ZoneId = _testZoneId,
+                Category = 0,
+                OnlyFree = true
+            };
```

Comment:
I don't see the difference between this Arrange block and the one in the test above.